### PR TITLE
Channel permissions model types changes

### PR DESCRIPTION
**Pull request by fowczarek**
- Extend `Group` model with `channels` and `restricted_access_to_channels` fields
- Update `User` type with `restictedAccessToChannels` and `channels` fields
  - `restictedAccessToChannels` - returns `False` if at least one user group has `restictedAccessToChannels` set to `False`, otherwise `True
  - `channels` - returns the sum of use group channels, if if at least one user group has `restictedAccessToChannels` set to `False`, all channels are returned. If user do not have any group, empty list is returned
- Update `Group` type with `restictedAccessToChannels` and `channels` fields

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-7880 Channel permissions [Orders]](https://app.clickup.com/t/2549495/SALEOR-7880)

File: saleor/graphql/account/types.py [hunk 1]
```diff
@@ -440,6 +462,18 @@ def resolve_permission_groups(root: models.User, _info: ResolveInfo):
...
+    @staticmethod
+    def resolve_accessible_channels(root: models.Group, info: ResolveInfo):
+        # Sum of channels from all user groups. If at least one group has
+        # `restrictedAccessToChannels` set to False - all channels are returned
+        return AccessibleChannelsByUserIdLoader(info.context).load(root.id)
+
+    @staticmethod
+    def resolve_restricted_access_to_channels(root: models.Group, info: ResolveInfo):
+        # Returns False if at least one user group has `restrictedAccessToChannels`
+        # set to False
```
[the remark this answers is not in the document]
Comment (IKarbowiak) [reply saleor/graphql/account/types.py:474]: Good idea!

[the remark this answers is not in the document]
Comment (IKarbowiak) [reply saleor/graphql/account/types.py:477]: done

File: saleor/graphql/account/dataloaders.py [hunk 2]
```diff
@@ -80,3 +81,104 @@ def batch_load(self, keys):
             .in_bulk(field_name="codename")
         )
         return [permission_map.get(codename) for codename in keys]
+
+
+class BaseAccessibleChannels(DataLoader):
+    abstract = True
```
Comment (kadewu) [inline saleor/graphql/account/dataloaders.py:87]: What do you think about using `ABC` to mark this as abstract?

**Review by kadewu — APPROVED** [review]

File: saleor/graphql/account/types.py [hunk 3]
```diff
@@ -440,6 +465,18 @@ def resolve_permission_groups(root: models.User, _info: ResolveInfo):
     def resolve_editable_groups(root: models.User, _info: ResolveInfo):
         return get_groups_which_user_can_manage(root)
 
+    @staticmethod
+    def resolve_accessible_channels(root: models.Group, info: ResolveInfo):
+        # Sum of channels from all user groups. If at least one group has
```
Comment (korycins) [inline saleor/graphql/account/types.py:470]: Question: Wouldn't be good to use this part to extend the field description?

File: saleor/graphql/account/types.py [hunk 4]
```diff
@@ -643,6 +680,18 @@ class Group(ModelObjectType[models.Group]):
...
     )
+    accessible_channels = NonNullList(
+        Channel,
+        description="List of channels the group has access to."
+        + ADDED_IN_313
+        + PREVIEW_FEATURE,
+    )
+    restricted_access_to_channels = graphene.Boolean(
+        required=True,
+        description="Determine if group have restricted access to channels."
```
Comment (korycins) [inline saleor/graphql/account/types.py:691]: If this is false, then we have an access to all channels, right? If yes, wouldn't be good to include this in the field description?

**Review by korycins — APPROVED** [review]

File: saleor/graphql/account/types.py (hunk 4 above)
Comment (IKarbowiak) [reply saleor/graphql/account/types.py:691]: No, this field will be defined directly in the `PermissionGroup` mutations

File: saleor/graphql/account/types.py (hunk 3 above)
Comment (IKarbowiak) [reply saleor/graphql/account/types.py:472]: Done

**Pull request merged**